src/ll/reply.rs

```diff
@@ -17,57 +17,77 @@ use zerocopy::AsBytes;
 const INLINE_DATA_THRESHOLD: usize = size_of::<u64>() * 4;
 pub(crate) type ResponseBuf = SmallVec<[u8; INLINE_DATA_THRESHOLD]>;
 
+pub(crate) enum ResponseData<'a> {
+    Error(i32),
+    Data(&'a [u8])
+}
+
+pub(crate) trait ResponseTrait {
+    fn get<'a>(&'a self) -> ResponseData<'a>;
+}
```

Comment:
Having `ResponseData`, `ResponseTrait`, `ResponseBuf`, and `Response` is more complex than I would like. Is it possible to add something like a `DataRef(&[u8])` to `Response` instead of this additional trait and enum?

Reply:
The problem with adding a `DataRef(&[u8])` is that it introduces a lifetime in `Response`. It makes things more complex when `Response` is used as a "owner" of the data (as in [dispatch_req](https://github.com/cberner/fuser/blob/c05bea5817fe201467a2546a4be3718bb60b46f3/src/request.rs#L68))

Another way to simplify this is to move the constrution of the `abi::fuse_*_out` in the structure themselves (with a `new`) and implement `ResponseTrait` for `ResponseBuf`. This way we could simply remove `Response` (and rename `ResponseTrait` to `Response` ?)

Reply:
I had a version of this change that added the `&[u8]` variant to `Response`, and it wasn't so bad: https://github.com/jamesbornholt/fuser/commit/51ca95dd4e469b9e16ce7588b1678570ff1dd03b. I was only trying to optimize the copy out of `ReplyData`, though—I didn't look at whether it was possible to avoid copying other reply types.

Reply:
Ya, something like that looks cleaner to me

Reply:
It indeed works and is simpler. Thanks for the other PR and have merged it.